mrmustard/lab/representations/converter.py

```diff
@@ -0,0 +1,497 @@
+
+        N = cov.shape[-1] // 2
+        sigma = self._pq_to_aadag(cov)
+        beta = self._pq_to_aadag(means)
+        Q = sigma + 0.5 * math.eye(2 * N, dtype=sigma.dtype)
+
+        return Q, beta
+
+    ########################################################################
+    ###                    From Wigner to Bargmann                       ###
+    ########################################################################
+
+    def _cayley(self, X: Tensor, c: float) -> Tensor:
+        r"""
+        Returns the Cayley transformation of a matrix:
+        :math:`cay(X) = (X - cI)(X + cI)^{-1}`
+
+        Args:
+            c (float): the parameter of the self._cayley transform
+            X (Tensor): a matrix
+
+        Returns:
+            The cayley transformation of X tensor with respect to c
+        """
+
+        I = math.eye(X.shape[0], dtype=X.dtype)
+
+        return math.solve(X + c * I, X - c * I)  # TODO : solve the argument nb issue
+
+    def _wignerdm_to_bargmanndm(self, wigner_dm: WignerDM) -> BargmannDM:
+        r"""
+        Converts the wigner representation in terms of covariance matrix and mean vector into the
+        Bargmann `A,B,C` triple for a density matrix (i.e. for `M` modes, `A` has shape `2M x 2M`
+        and `B` has shape `2M`).
+        The order of the rows/columns of A and B corresponds to a density matrix with the usual
+        ordering of the indices.
+
+        Note that here A and B are defined with respect to the literature.
+
+        Args:
+            wigner_dm (WignerDM) : the Wigner DM representation of the state
+
+        Returns:
+            The BargmannDM representation of the input state
+
+        """
+        N = wigner_dm.data.cov.shape[-1] // 2
+
+        A = math.matmul(math.Xmat(N), self._cayley(self._pq_to_aadag(wigner_dm.data.cov), c=0.5))
+        Q, beta = self._wigner_to_husimi(wigner_dm.data.cov, wigner_dm.data.means)
+        b = math.solve(Q, beta)
+        B = math.conj(b)
+        C = math.exp(-0.5 * math.sum(math.conj(beta) * b)) / math.sqrt(math.det(Q))
+
+        return BargmannDM(A=A, b=B, c=C)
+
+    def _wignerket_to_bargmannket(self, wigner_ket: WignerKet) -> BargmannKet:
+        r"""
+        Converts the Wigner representation in terms of covariance matrix and mean vector into the
+        Bargmann A,B,C triple for a Hilbert vector (i.e. for M modes, A has shape M x M and B has
+        shape M).
+
+        Args:
+            wigner_ket (WignerKet) : the input Wigner representation of the state
+
+        Returns:
+            The Bargmann representation of the input state
+        """
+
+        N = wigner_ket.data.symplectic.shape[-1] // 2
+        cov = wigner_ket.cov
+        means = wigner_ket.means
+        wigner_dm = WignerDM(cov=cov, means=means)
+        bargmann_dm = self._wignerdm_to_bargmanndm(wigner_dm=wigner_dm)
+
+        return BargmannKet(
+            bargmann_dm.data.A[N:, N:], bargmann_dm.data.b[N:], math.sqrt(bargmann_dm.data.c)
```

Comment:
wigner ket has no global phase information to preserve here?

Reply:
Yes, we miss that information from the wignerdm to bargmanndm and it asked for extra codes to get the global phase part. Are we going to add it?

Reply:
if it's not too much work, yes, it would be good to have it